### PR TITLE
Adds Plan Comparison view

### PR DESCRIPTION
**Pull request by frosty**
Implements #4799, adding a comparison view for plans.

### Known issues
- No state restoration – we'll revisit that after this has been merged.
- The current active plan isn't highlighted, as that requires quite a visual change from the current plan detail screens. Will raise a separate issue to track that.

Needs review: @koke

### CODE REVIEW

**Comment by koke** [comment]
When you tap on one of the dots, it doesn't change page, and I think it should. What I understand from the docs is that it's what `pageControlChanged` is supposed to be doing, but it's not being called for me.

**Comment by frosty** [comment]
Huh, it's definitely supposed to, and it definitely _was_ working. I'll take a look.

**Comment by frosty** [comment]
Ah, the page control just wasn't big enough. I've updated its constraints so it actually fills the bottom area (you can now tap anywhere in the bottom area to page left or right). I had been testing on a device and actually I think it was easier to trigger with my inaccurate fingers than with the mouse.

File: WordPress/Classes/ViewRelated/Plans/PlanComparisonViewController.swift [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+        guard let index = currentIndex else { return nil }
+        
+        return viewControllerAtIndex(index + 1)
+    }
+    
+    func pageViewController(pageViewController: UIPageViewController, willTransitionToViewControllers pendingViewControllers: [UIViewController]) {
+        if let controller = pendingViewControllers.first as? PlanDetailViewController {
+            currentPlan = allPlans[indexOfViewController(controller)]
+        }
+    }
```
Comment (koke) [inline WordPress/Classes/ViewRelated/Plans/PlanComparisonViewController.swift]: I think we should drop this one. It felt a bit odd that the title and page control updated as soon as you start dragging. If you look at the iOS home screen, the page control only updates when the transition finishes.

File: WordPress/Classes/ViewRelated/Plans/PlanComparisonViewController.swift [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+    
+    /// `success` is true if the scrolling action was successful (i.e. there was a page to scroll to)
+    private func scrollToPreviousPage(completion: ((success: Bool) -> ())? = nil) {
+        guard let previousViewController = previousViewController() else {
+            completion?(success: false)
+            return
+        }
+        
+        pageViewController.setViewControllers([previousViewController], direction: .Reverse, animated: true, completion: nil)
+        completion?(success: false)
```
Comment (koke) [inline WordPress/Classes/ViewRelated/Plans/PlanComparisonViewController.swift]: I think this should be `true`. It's acting weird when you tap on the page control to go left.

File: WordPress/Classes/ViewRelated/Plans/PlanComparisonViewController.swift [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+    }
+}
```
Comment (koke) [inline WordPress/Classes/ViewRelated/Plans/PlanComparisonViewController.swift]: Can you add a new line at the end of the file?

File: WordPress/Classes/ViewRelated/Plans/PlanComparisonViewController.swift (hunk 2 above)
Comment (frosty) [inline WordPress/Classes/ViewRelated/Plans/PlanComparisonViewController.swift]: Thanks for spotting this!

File: WordPress/Classes/ViewRelated/Plans/PlanComparisonViewController.swift (hunk 1 above)
Comment (frosty) [inline WordPress/Classes/ViewRelated/Plans/PlanComparisonViewController.swift]: Yep, much better.

File: WordPress/Classes/ViewRelated/Plans/PlanComparisonViewController.swift (hunk 3 above)
Comment (frosty) [inline WordPress/Classes/ViewRelated/Plans/PlanComparisonViewController.swift]: :+1:

**Comment by koke** [comment]
There's this weird issue with the page control:
- Open the comparison view
- Tap on the right side of the page control so it changes page
- Immediately tap and hold on the list, trying to make it stay on the same page
- The current page stays there but the page control and title changes

I've been messing with the code and passed the completion on `scrollToNextPage` to `setViewControllers`, but it's being called with `success == true` even if you cancel the animation by "holding" the view in place. I'm not sure if it's fixable with the current API.

I'll let you try fix it, otherwise :shipit:

**Comment by frosty** [comment]
@koke I've converted this VC to use a scrollview instead of a pageViewController to get around some of the weirdness we were seeing with a custom page control. Could you give it another check over for me please? If you could also ensure you verify on iPad (where it's displayed in a modal), in both orientations and in multitasking, as well as voiceover that'd be great.

Thank you!

File: WordPress/Classes/ViewRelated/Plans/PlanComparisonViewController.swift [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+    }
+    
+    private func currentScrollViewPage() -> Int {
+        // Calculate which plan's VC is at the center of the view
+        let pageWidth = scrollView.bounds.width
+        let centerX = scrollView.contentOffset.x + (pageWidth / 2)
+        let currentPage = Int(floor(centerX / pageWidth))
+
+        // Keep it within bounds
+        return max(min(currentPage, allPlans.count - 1), 0)
```
Comment (koke) [inline WordPress/Classes/ViewRelated/Plans/PlanComparisonViewController.swift]: See `Int.clamp(min:max:)` ;)

**Comment by koke** [comment]
:shipit:

**Pull request merged**